design-documents/promises.md

```diff
@@ -0,0 +1,240 @@
+### Why?
+Service contracts in the future will often be executed in an asynchronous manner
```

Comment:
Please elaborate more on the purpose and provide examples where it can be used now. As I understand, the purpose is to allow call APIs in parallel when we need to call multiple APIs consequentially and as a result improve performance?

Reply:
public API methods don't have to return promise since we'll be calling them via HTTP. If you mean that when a service contract implementation will actually be calling web API on another server then yes, if we go with the 1st route they'd have to return a promise